src/soundvectors.py

```diff
@@ -1,317 +1,634 @@
-        "round": 0, "velaric": 0, "long": 0, "ant": 0, "distr": 0, "strid": 0, }}, "raising": {
-    "lowered": {"cons": 0, "syl": 0, "son": 0, "cont": 0, "delrel": 0, "lat": 0, "nas": 0, "voi": 0, "sg": 0, "cg": 0,
-        "pharyngeal": 0, "laryngeal": 0, "cor": 0, "dorsal": 0, "lab": 0, "hi": 0, "lo": 1, "back": 0, "round": 0,
-        "velaric": 0, "long": 0, "ant": 0, "distr": 0, "strid": 0, },
-    "raised": {"cons": 0, "syl": 0, "son": 0, "cont": 0, "delrel": 0, "lat": 0, "nas": 0, "voi": 0, "sg": 0, "cg": 0,
-        "pharyngeal": 0, "laryngeal": 0, "cor": 0, "dorsal": 0, "lab": 0, "hi": 1, "lo": 0, "back": 0, "round": 0,
-        "velaric": 0, "long": 0, "ant": 0, "distr": 0, "strid": 0, }, }, "relative_articulation": {
-    "centralized": {"cons": 0, "syl": 0, "son": 0, "cont": 0, "delrel": 0, "lat": 0, "nas": 0, "voi": 0, "sg": 0,
-        "cg": 0, "pharyngeal": 0, "laryngeal": 0, "cor": 0, "dorsal": 0, "lab": 0, "hi": 0, "lo": 0, "back": 0,
-        "round": 0, "velaric": 0, "long": 0, "ant": 0, "distr": 0, "strid": 0, },
-    "mid-centralized": {"cons": 0, "syl": 0, "son": 0, "cont": 0, "delrel": 0, "lat": 0, "nas": 0, "voi": 0, "sg": 0,
-        "cg": 0, "pharyngeal": 0, "laryngeal": 0, "cor": 0, "dorsal": 0, "lab": 0, "hi": 0, "lo": 0, "back": 0,
-        "round": 0, "velaric": 0, "long": 0, "ant": 0, "distr": 0, "strid": 0, },
-    "advanced": {"cons": 0, "syl": 0, "son": 0, "cont": 0, "delrel": 0, "lat": 0, "nas": 0, "voi": 0, "sg": 0, "cg": 0,
-        "pharyngeal": 0, "laryngeal": 0, "cor": 0, "dorsal": 0, "lab": 0, "hi": 0, "lo": 0, "back": 0, "round": 0,
-        "velaric": 0, "long": 0, "ant": 0, "distr": 0, "strid": 0, },
-    "retracted": {"cons": 0, "syl": 0, "son": 0, "cont": 0, "delrel": 0, "lat": 0, "nas": 0, "voi": 0, "sg": 0, "cg": 0,
-        "pharyngeal": 0, "laryngeal": 0, "cor": 0, "dorsal": 0, "lab": 0, "hi": 0, "lo": 0, "back": 0, "round": 0,
-        "velaric": 0, "long": 0, "ant": 0, "distr": 0, "strid": 0, }, },
-    "centrality": {"back": {"back": 1}, "central": {}, "front": {"front": 1}, "near-back": {"back": 1},
-        "near-front": {"front": 1}, }, "rhotacization": {"rhotacized": {}},
-    "height": {"close": {"hi": 1, "tense": 1}, "close-mid": {"tense": 1}, "mid": {}, "near-close": {"hi": 1},
-        "near-open": {"lo": 1}, "open": {"lo": 1, "tense": 1}, "open-mid": {}, },
-    "friction": {"with-friction": {"son": -1}}, "roundedness": {"rounded": {"round": 1}, "unrounded": {"round": -1}},
-    "rounding": {"less-rounded": {"round": 1}, "more-rounded": {"round": 1}},
-    "start": {"from-high": {"hireg": 1, "hitone": 1}, "from-low": {"loreg": 1}, "from-mid": {},
-        "from-mid-high": {"hireg": 1}, "from-mid-low": {"loreg": 1, "hitone": 1}, },
-    "contour": {"contour": {"contour": 1}, "falling": {"falling": 1}, "rising": {"rising": 1}, "flat": {},
-        "short": {}, },
-    "to_roundedness": {"to_rounded": {"secondrounded": 1}, "to_unrounded": {"secondrounded": -1}, }, }
+clts_features = {
```

Comment:
Sorry for the many changed lines - but I had to reformat this thing to get an idea of the keys, and which keys appear on which nesting level ...

Reply:
No need to apologize - looks much better like that!

Reply:
I often fall into the trap of mixing code style changes with "meaningful" changes when creating PRs for review. So to keep me from falling into this trap again, I have to acknowledge it at least :)

Reply:
I started to compare them with the keys in the `clts_features`, but stopped putting them on one line when I understood how they are used. My "real" suggestion for `binary_features` would not be different formatting, but turning the list into an `enum.Enum` - that way you'd turn typos into something that an IDE like PyCharm could spot and warn about. `StrEnum` would be ideal for this - but only came into Python at 3.11...

Reply:
On second thought: What about replacing the `FeatureBundle` class with a `dataclasses.dataclass`?
```python
import dataclasses

@dataclasses.dataclass
class FeatureBundle:
    cons: int
    syl: int
    ...
```
Then we'd have `binary_features` simply as `[f.name for f in dataclasses.fields(FeatureBundle)]`, `vec_dict` as `vars(...)` and `vector` would be `dataclasses.astuple(...)`.

Reply:
Oh, and in the long `dict` defining the default values for feature domains, the default feature bundles could then be instances of the dataclass, possibly only defining the non-zero features.

Reply:
@arubehn let me know, if I should take a stab at refactoring `FeatureBundle` into a dataclass. So far, I usually went with the `attrs` package for this stuff. But I think dataclasses could be perfect here.

Reply:
I think this would make a lot of sense. Sorry, I saw these discussions only now. Somehow I missed the email reminder of the PR and was too much away from my desk yesterday.

Reply:
BTW: for the formatting, I used pycharm's automated formatting to some dense format last time, after having experimented with the automated formatting by black that puts everything on a different line. So one can get easily back to that later.

Reply:
I just checked how this would "feel" in dataclasses, and I think if I understand you properly, @xrotwang, you suggest to replace also the long dictionaries by FeatureBundle instances? So we could use a dictionary in the form of:

```python

@dataclass(order=True)
class FeatureBundle:
    cons: int = 0
    syl: int = 0
    cont: int = 0
    ...
    cg: int = 0
    ...

clts_feautures = {
    ...
    "ejection": FeatureBundle(cg=1),
}
```

and with the dataclass methods `asdict` and `astuple`, we could even directly turn these into an ordered dict or a vector. I Think this would make it also much easier to maintain the databasis, since it would reduce the large dictionaries.

Reply:
So I definitely support this solution.